manifests/init.pp

```diff
@@ -44,11 +45,12 @@
   Enum['refpolicy', 'simple'] $default_builder                = 'simple',
 
   ### START Hiera Lookups ###
-  $boolean        = undef,
-  $fcontext       = undef,
-  $module         = undef,
-  $permissive     = undef,
-  $port           = undef,
+  Optional[Hash] $boolean         = undef,
```

Comment:
Would it make sense (and work) to just make this (and the other ones below it) ```Hash $boolean = {}``` and still use hiera?

Reply:
Shouldn't be a problem and i prefer to have simple datatypes too. But it would break for users explicitly passing "undef" to the class (to override hiera values from higher up) which would no longer be valid (no idea if anyone actually does this).

Reply:
That's a valid point

Reply:
does passing undef work to not trigger automatic parameter lookup? I think there was a discussion about that some time ago. 

But I prefer undef over an empty hash as it shows that nothing needs to be passed and Optional describes this.